### DISCUSSION TITLE
Add attributes to KML exports

### DISCUSSION THREAD

**Discussion by PBoissier** [IPGP/webobs#134] [Ideas]
Hello everyone,

I have a request from colleagues for exports in KML format: the ability to add attributes to exported NODES. In particular the date of installation and end of the NODE.
This would make it possible to import the KMLs into a GIS tool like QGIS and simply make network maps by selecting, for example, only the stations that are still active.

It would consist of adding an **ExtendedData** section to the **Placemark**.

For example, the **Placemark** for the RVRG GNSS NODE:

```
  <Placemark id="RDCRVRG">
    <name>RVRG : "Rivals"</name>
    <description><![CDATA[<i></i><br>Déformations / GNSS OVPF<br><small>(VIEW.GNSS.RDCRVRG)</small>]]></description>
    <open>1</open>
    <styleUrl>#webobs</styleUrl>
    <Point><coordinates>55.711555,-21.253548,2240.0</coordinates></Point>
  </Placemark>
```

Could become :

```
  <Placemark id="RDCRVRG">
    <name>RVRG : "Rivals"</name>
    <ExtendedData>
      <Data name="active">
        <value>0</value>
      </Data>
      <Data name="start">
        <value>2001-01-01</value>
      </Data>
      <Data name="end">
        <value>2004-01-01</value>
      </Data>
    </ExtendedData>
    <description><![CDATA[<i></i><br>Déformations / GNSS OVPF<br><small>(VIEW.GNSS.RDCRVRG)</small>]]></description>
    <open>1</open>
    <styleUrl>#webobs</styleUrl>
    <Point><coordinates>55.711555,-21.253548,2240.0</coordinates></Point>
  </Placemark>
```

I located the section of code to modify in _**nloc.pl**_ and I can submit this evolution soon.

Before I start making an edit, is it OK for you? Would you see any other attributes to add? (the type of the NODE, its alias, etc.)

**Comment by beaudu** [comment]
Hi,
Thanks for the suggestion and great improvement of KML export. It is OK to submit this modification. If you find any other metadata that can be exported in the KML format it will be ok also.
F.
